### PR TITLE
privilege: fix DBIsVisible privilege check

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve? 
Originally, user with any one of global privileges can see all database names and use them.
But MySQL just support few global privileges to access any databases.

### What is changed and how it works?
fix privilege check for global privilege in `DBIsVisible`

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variable/fields change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch

Release note

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -37,11 +37,11 @@ require (
 	github.com/pingcap/goleveldb v0.0.0-20171020122428-b9ff6c35079e
 	github.com/pingcap/kvproto v0.0.0-20191217072959-393e6c0fd4b7
 	github.com/pingcap/log v0.0.0-20191012051959-b742a5d432e9
-	github.com/pingcap/parser v0.0.0-20191223023445-b93660cf3e4e
+	github.com/pingcap/parser v0.0.0-20191223211552-4cf10469de1d
```
Comment (tiancaiamao) [inline go.mod]: Did you run 

```
go get github.com/pingcap/parser@master
```

Why many other irrelevant packages are changed?

File: privilege/privileges/cache.go [hunk 2]
```diff
@@ -909,7 +909,7 @@ func (p *MySQLPrivilege) RequestVerification(activeRoles []*auth.RoleIdentity, u
 // DBIsVisible checks whether the user can see the db.
 func (p *MySQLPrivilege) DBIsVisible(user, host, db string) bool {
 	if record := p.matchUser(user, host); record != nil {
-		if record.Privileges != 0 {
+		if record.Privileges&mysql.GlobalDBVisible > 0 {
```
Comment (tiancaiamao) [inline privilege/privileges/cache.go]: Actually we can define `GlobalDBVisible` in the TiDB repo if it's not already added to the parser repo.

File: go.mod (hunk 1 above)
Comment (imtbkcat) [reply go.mod]: yes, i added `GlobalDBVisible` into parser repo and run go get.

Comment (tiancaiamao) [reply go.mod]: github.com/pingcap/tipb is updated, that's weird

**Comment by tiancaiamao** [comment]
LGTM

**Comment by imtbkcat** [comment]
conflict has been resolved, PTAL @zz-jason

**Pull request merged**